spring-cloud-gcp-autoconfigure/src/test/java/com/google/cloud/spring/autoconfigure/sql/R2dbcCloudSqlEnvironmentPostProcessorTests.java

```diff
@@ -100,18 +103,44 @@ void testSetR2dbcProperty_mySql_urlProvidedByUserIgnored() {
   }
 
   @Test
-  void testSetR2dbcProperty_postgres_defaultUsername() {
+  void testSetR2dbcProperty_postgres() {
+    validateDatabaseProperties(new String[] {"io.r2dbc.postgresql"},
+        "postgres",
+        "r2dbc:gcp:postgres://my-project:region:my-instance/my-database");
+  }
+
+  @Test
+  void testSetR2dbcProperty_mysql() {
+    validateDatabaseProperties(new String[] {"dev.miku.r2dbc.mysql"},
+        "root",
+        "r2dbc:gcp:mysql://my-project:region:my-instance/my-database");
+  }
+
+  /**
+   * Accepts a list of packages to _keep_, and tests that correct properties for the database type
+   * got injected into context.
+   * The input is used to calculate the inverse list of packages to _remove_ that is passed to
+   * {@code FilteredClassLoader}
+   *
+   * @param includePackages a list of driver packages to keep on the classpath
+   */
+  private void validateDatabaseProperties(String[] includePackages, String username, String url) {
```

Comment:
How about we name this to `verifyhatCorrectUrlAndUsernameSet`? Something that tells us what the expected result is.